lib/services/storage_service.ts

```diff
@@ -293,7 +296,7 @@ export class SNStorageService extends PureService {
    */
   async clearValues() {
     this.setInitialValues();
-    await this.repersistToDisk();
+    await this.immediatelyPersistValuesToDisk();
```

Comment:
The one difference I can find between the old and new implementations is that this new placement in `clearValues` does not execute the persistence in a `executeCriticalFunction` block. I can't recall the specifics of timing here, but to be safe, we should perhaps move the `executeCriticalFunction` call from `persistValuesToDisk` to `immediatelyPersistValuesToDisk`. This way, we know it's always executed as a critical function.

Reply:
Good point. I've pushed changes to address this.
Let me know if you have a better idea for the method's name, I've really been struggling with this one.

Reply:
I think it's fine. Typically with this pattern I go with persistValuesToDisk/performPersistValuesToDisk. But in this case since we have a todo to debounce the former function, the "immediately" makes sense.